### PR TITLE
feat: add findSubMatch behavior

### PR DESCRIPTION
**Pull request by smerle33**
As per : https://github.com/updatecli/updatecli/pull/551#issuecomment-1058903823
Adding information on findSubMatch.

### CODE REVIEW

File: content/docs/core/transformer.adoc [hunk 1]
```diff
@@ -86,6 +100,9 @@ sources:
       - replacers:
           - from: "-jdk15"
             to: "-jdk17"
+      - findsubmatch:
+          pattern: '"(.*?):'
```
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:104]: I would use an example with multiple possible sub matches here.

Comment (dduportal) [reply content/docs/core/transformer.adoc:104]: Then 2 examples :)

File: content/docs/core/transformer.adoc [hunk 2]
```diff
@@ -66,6 +66,20 @@ A replacer rule modifies the information by replacing the "from" value by the "t
 | to | &#10004;|-| "to" value defines the string that we want to have
 |===
 
+
+
+=== FindSubMatch
+
+A findSubMatch rule capture the information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).
```
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:73]:
```suggestion
A findSubMatch rule captures the desired information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).
```

File: content/docs/core/transformer.adoc (hunk 1 above)
Comment (smerle33) [reply content/docs/core/transformer.adoc]: I updated this one, 2 exemples are not compliant with what's existing

**Comment by olblak** [comment]
Thanks @smerle33 for the documentation \o/
I'll merge this pullrequest once we trigger the next updatecli release

File: content/docs/core/transformer.adoc [hunk 3]
```diff
@@ -66,6 +66,21 @@ A replacer rule modifies the information by replacing the "from" value by the "t
...
+=== FindSubMatch
+
+
+A findSubMatch rule captures the desired information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).
+
+[cols="1,1,1,4",options=header]
+|===
+| Name | Required | Default |Description
+| pattern | &#10004;|-| "pattern" value defines the regular expression to apply
+| captureIndex |-|-| "captureIndex" value defines the index of the capture to return (0 by default)
```
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:80]: As discussed together, add explanation about "0" returning all submatches, and individual submatch indexes starting at 1

(In the example below, `captureIndex: 1` returns the `(.*?)` submatch)

File: content/docs/core/transformer.adoc [hunk 4]
```diff
@@ -86,6 +101,9 @@ sources:
       - replacers:
           - from: "-jdk15"
             to: "-jdk17"
+      - findsubmatch:
+          pattern: '"(.*?):(.*)v'
```
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:105]:
```suggestion
          pattern: '"(.*?):(.*)'
```

**Review by lemeurherve — CHANGES_REQUESTED** [review]

File: content/docs/core/transformer.adoc (hunk 3 above)
Comment (smerle33) [reply content/docs/core/transformer.adoc:80]: awesome but I don't have the button to commit the suggestion ...

Comment (lemeurherve) [reply content/docs/core/transformer.adoc:80]: It's a simple comment, not a suggestion, you'll need to go to the "files" tab and create a new suggestion from it, or copy/paste it in your local repo and commit then push it.

Note: this should also be added as comment in updatecli/updatecli corresponding code

File: content/docs/core/transformer.adoc [hunk 5]
```diff
@@ -66,6 +66,21 @@ A replacer rule modifies the information by replacing the "from" value by the "t
...
+
+[cols="1,1,1,4",options=header]
+|===
+| Name | Required | Default |Description
+| pattern | &#10004;|-| "pattern" value defines the regular expression to apply
+| captureIndex |-|-| "captureIndex" value defines the index of the capture to return (0 by default)
+|===
+
+
 == Example
```
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:85]:
```suggestion
The `findSubmatch` below will return the submatch captured with `(.*?)`:
```

File: content/docs/core/transformer.adoc (hunk 3 above)
Comment (lemeurherve) [inline content/docs/core/transformer.adoc:80]:
```suggestion
| captureIndex |-|-| "captureIndex" value defines the index of the capture to return. Note that the default value `0` returns all submatches, and individual submatch indexes start at `1`.
```

Comment (lemeurherve) [reply content/docs/core/transformer.adoc:80]: @smerle33 I've added the suggestions and the PR ;)

**Review by lemeurherve — APPROVED** [review]

**Review by olblak — APPROVED** [review]

**Pull request merged**